cobra/flux_analysis/deletion.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import multiprocessing
+import logging
+from warnings import warn
+from itertools import product
+from operator import attrgetter
+from collections import defaultdict
+from builtins import (map, dict)
+from future.utils import raise_
+
+from ..manipulation.delete import (find_gene_knockout_reactions)
+import cobra.util.solver as sutil
+
+try:
+    import scipy
+except ImportError:
+    moma = None
+else:
+    from . import moma
+
+try:
+    from pandas import DataFrame
+except:
+    DataFrame = None
+
+LOGGER = logging.getLogger(__name__)
+
+
+def infinite_defaultdict():
+    return defaultdict(infinite_defaultdict)
+
+
+def defaultdict_to_dict(def_dict):
+    if not isinstance(def_dict, defaultdict):
+        return def_dict
+    for k, v in def_dict.items():
+        def_dict[k] = defaultdict_to_dict(v)
+    return dict(def_dict)
+
+
+def _reactions_knockouts_with_restore(model, reactions):
+    to_restore = {}
+    growth = None
+    for reaction in reactions:
+        to_restore[reaction] = reaction.bounds
+        reaction.knock_out()
+    try:
+        solution = model.optimize()
+        if solution.status != 'infeasible':
+            growth = solution.objective_value
+    except:
+        pass
+    for reaction in reactions:
+        reaction.bounds = to_restore[reaction]
+    return (growth, [r.id for r in reactions])
+
+
+def _reaction_deletion_worker(ids):
+    model = _reaction_deletion_worker.model
+    return _reactions_knockouts_with_restore(
+        model,
+        [model.reactions.get_by_id(r_id) for r_id in ids]
+    )
+
+
+def _gene_deletion_worker(ids):
+    model = _gene_deletion_worker.model
+    all_reactions = []
+    for g_id in ids:
+        all_reactions.extend(find_gene_knockout_reactions(model, (model.genes.get_by_id(g_id),)))
```

Comment:
This will be slow. Preferably `find_gene_knockout_reactions` should be called once before the map step with all genes that are knocked out.